### ISSUE TITLE
[bst-1] checks for push in a pull only remote

### ISSUE THREAD

**Issue by abderrahim** [apache/buildstream#1724]
bst 1 tries to check whether the remote supports asset push even if the remote is not configured for push. This combined with a grpcio bug (that produces weird errors when getting http 404) means that initializing remotes fails with an error like

```
[--:--:--][][] WARNING Failed to initialize remote https://cache.freedesktop-sdk.io:11001: Stream removed
```

**Comment by gtristan** [comment]
This is surprising, you'd think we would notice this sooner, since pull-only remotes are used with great frequency.

Can you check if this is fixed by #1725 ?

I'm curious where the message "Stream removed" comes from.

**Comment by abderrahim** [comment]
Ok, something I forgot to mention here: it only happens with `grpcio >= 1.46` and that's why I think it's a bug in grpcio that exposes it.

**Comment by gtristan** [comment]
This is apparently fixed by #1725
